### PR TITLE
Documentation around process-auditing

### PR DESCRIPTION
**Pull request by directionless**
Added additional documentation that`--disable_audit=false` is required for macOS.

Added additional debugging for events.

### CODE REVIEW

File: docs/wiki/deployment/process-auditing.md [hunk 1]
```diff
@@ -2,14 +2,75 @@
...
+| disable_audit  | bool | Disable receiving events from the audit subsystem | true          | false | 0          |
+| disable_events | bool | Disable osquery publish/subscribe system          | false         | false | 0          |
++----------------+------+---------------------------------------------------+---------------+-------+------------+
+```
+
+### Examine event table
+
+osquery keeps state about the events subsystem in the `osquery_events`
+table. The `subscriptions` is of note here. This example is from a
+macOS machine with events enabled. (But no events detected)
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: It's not clear what the troubleshooting goal is here. Perhaps something additional like:

"At this point you should attempt to trigger an event. Then confirm that the event count is non-0."

File: docs/wiki/deployment/process-auditing.md [hunk 2]
```diff
@@ -2,14 +2,75 @@
...
+| file_events             | fsevents        | subscriber | 0             | 0      | 0         | 1      |
+| hardware_events         | iokit           | subscriber | 1             | 0      | 0         | 1      |
+| process_events          | openbsm         | subscriber | 8             | 0      | 0         | 1      |
+| user_events             | openbsm         | subscriber | 1             | 0      | 0         | 1      |
+| user_interaction_events | event_tapping   | subscriber | 1             | 0      | 0         | 1      |
+| yara_events             | fsevents        | subscriber | 0             | 0      | 0         | 1      |
++-------------------------+-----------------+------------+---------------+--------+-----------+--------+
+```
+
+### Try to get working on osqueryi
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: nitpick, consider rephrasing

File: docs/wiki/deployment/process-auditing.md [hunk 3]
```diff
@@ -2,14 +2,75 @@
...
+
+You can run osqueryi and see events. This can be valuable in working
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: nitpick, inconsistent `osqueryi` and "osqueryi"

File: docs/wiki/deployment/process-auditing.md (hunk 2 above)
Comment (theopolis) [reply docs/wiki/deployment/process-auditing.md]: Additionally, I think this section could be dropped or used as an introduction with general troubleshooting.

File: docs/wiki/deployment/process-auditing.md [hunk 4]
```diff
@@ -101,7 +162,9 @@ On Linux, a companion table called `user_events` is included that provides sever
 
 ## macOS process auditing
 
-osquery has support for OpenBSM audit on Darwin platforms. This feature is already enabled on all macOS installations but doesn't audit process execution or the root user with default settings. To start process auditing on macOS, edit the `audit_control` file in `/etc/security/`. An example configuration is provided below but the important flags are: `ex`, `pc`, `argv`, and `arge`. The `ex` flag will log `exec` events while `pc` logs `exec`, `fork`, and `exit`. If you don't need `fork` and `exit` you may leave that flag out however in future, getting parent pid may require `fork`. If you care about getting the arguments and environment variables you also need `argv` and `arge`. More about these flags can be found [here](https://www.freebsd.org/cgi/man.cgi?apropos=0&sektion=5&query=audit_control&manpath=FreeBSD+7.0-current&format=html). Note that it might require a reboot of the system for these new flags to take effect. `audit -s` should restart the system but your mileage may vary.
+osquery has support for OpenBSM audit on Darwin platforms. To enable it in osquery, you need to set `--disable_audit=false`
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: "osquery supports OpenBSM auditing on macOS platforms."

File: docs/wiki/deployment/process-auditing.md [hunk 5]
```diff
@@ -101,7 +162,9 @@ On Linux, a companion table called `user_events` is included that provides sever
...
+
+OpenBSM is enabled on all macOS installations, but doesn't audit process execution or the root user with default settings. To start process auditing on macOS, edit the `audit_control` file in `/etc/security/`. An example configuration is provided below but the important flags are: `ex`, `pc`, `argv`, and `arge`. The `ex` flag will log `exec` events while `pc` logs `exec`, `fork`, and `exit`. If you don't need `fork` and `exit` you may leave that flag out however in future, getting parent pid may require `fork`. If you care about getting the arguments and environment variables you also need `argv` and `arge`. More about these flags can be found [here](https://www.freebsd.org/cgi/man.cgi?apropos=0&sektion=5&query=audit_control&manpath=FreeBSD+7.0-current&format=html). Note that it might require a reboot of the system for these new flags to take effect. `audit -s` should restart the system but your mileage may vary.
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: nitpick, avoid contractions. 

The linkified "here" should reference where they will go, "on FreeBSD's man page"

File: docs/wiki/deployment/process-auditing.md (hunk 1 above)
Comment (directionless) [reply docs/wiki/deployment/process-auditing.md]: I'm trying to give people the ability to diagnose what's happening. I'll try to reword

File: docs/wiki/deployment/process-auditing.md (hunk 2 above)
Comment (directionless) [reply docs/wiki/deployment/process-auditing.md]: agreed.

**Comment by directionless** [comment]
I reworded a bit. Maybe clearer now

File: docs/wiki/deployment/process-auditing.md [hunk 6]
```diff
@@ -2,14 +2,91 @@
...
+osquery> select * from osquery_flags where name in ("disable_events", "disable_audit");
++----------------+------+---------------------------------------------------+---------------+-------+------------+
+| name           | type | description                                       | default_value | value | shell_only |
++----------------+------+---------------------------------------------------+---------------+-------+------------+
+| disable_audit  | bool | Disable receiving events from the audit subsystem | true          | false | 0          |
+| disable_events | bool | Disable osquery publish/subscribe system          | false         | false | 0          |
++----------------+------+---------------------------------------------------+---------------+-------+------------+
+```
+
+
```
Comment (theopolis) [inline docs/wiki/deployment/process-auditing.md]: nitpick, remove triple-newline

**Comment by theopolis** [comment]
Mind rebasing onto master?

**Pull request merged**